langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/WanxImageModel.java

```diff
@@ -99,7 +100,14 @@ public Response<Image> edit(Image image, String prompt) {
 
         try {
             ImageSynthesisResult result = imageSynthesis.call(builder.build());
-            return Response.from(imagesFrom(result).get(0));
+            List<Image> images = imagesFrom(result);
+            if (images.isEmpty()) {
+                ImageSynthesisOutput output = result.getOutput();
+                String errorMessage = String.format("[%s] %s: %s",
+                        output.getTaskStatus(), output.getCode(), output.getMessage());
+                throw new NullPointerException(errorMessage);
```

Comment:
hey @jiangsier-xyz! curious if this case is really NPE. Maybe we can throw better exception here? thanks!

Reply:
> hey @jiangsier-xyz! curious if this case is really NPE. Maybe we can throw better exception here? thanks!

I've thought about this，but have no good idea. What are your suggestions?

Reply:
sorry for delayed response. I noticed project tend to use IllegalStateException for result errors

Reply:
> sorry for delayed response. I noticed project tend to use IllegalStateException for result errors

OK, that makes sense.